js_modules/dagit/packages/ui/src/components/TagSelector.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {Box} from './Box';
+import {Checkbox} from './Checkbox';
+import {Colors} from './Colors';
+import {Icon} from './Icon';
+import {MenuItem, Menu} from './Menu';
+import {Popover} from './Popover';
+import {Tag} from './Tag';
+
+type TagProps = {
+  close: (ev: React.SyntheticEvent<HTMLDivElement>) => void;
+};
+type DropdownItemProps = {
+  toggle: () => void;
+  selected: boolean;
+};
+type Props = {
+  placeholder?: React.ReactNode;
+  allTags: string[];
+  selectedTags: string[];
+  setSelectedTags: (tags: React.SetStateAction<string[]>) => void;
+  renderTag?: (tag: string, tagProps: TagProps) => React.ReactNode;
+  renderTagList?: (tags: React.ReactNode[]) => React.ReactNode;
+  renderDropdown?: (dropdown: React.ReactNode) => React.ReactNode;
+  renderDropdownItem?: (tag: string, dropdownItemProps: DropdownItemProps) => React.ReactNode;
+  dropdownStyles?: React.CSSProperties;
+};
+
+const defaultRenderTag = (tag: string, tagProps: TagProps) => {
+  return (
+    <Tag>
+      <Box flex={{direction: 'row', gap: 4, justifyContent: 'space-between', alignItems: 'center'}}>
+        <span>{tag}</span>
+        <Box style={{cursor: 'pointer'}} onClick={tagProps.close}>
+          <Icon name="close" />
+        </Box>
+      </Box>
+    </Tag>
+  );
+};
+
+const defaultRenderDropdownItem = (tag: string, dropdownItemProps: DropdownItemProps) => {
+  return (
+    <label>
+      <MenuItem
+        text={
+          <Box flex={{alignItems: 'center', gap: 8}}>
+            <Checkbox checked={dropdownItemProps.selected} onChange={dropdownItemProps.toggle} />
+            <span>{tag}</span>
+          </Box>
+        }
+      />
```

Comment:
I think `MenuItem` renders an `a` -- you probably want `tagName="div"` on the `MenuItem` to avoid ending up with a checkbox inside an anchor.

Reply:
good to kow